e2e/testcases/stress_test.go

```diff
@@ -261,6 +265,86 @@ func TestStressLargeRequest(t *testing.T) {
 	}
 }
 
+func TestStress100CRDs(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.Unstructured, ntopts.StressTest,
+		ntopts.WithReconcileTimeout(configsync.DefaultReconcileTimeout))
+
+	// Apply 100 small CRDs on the cluster to make sure Config Sync can still work.
+	for i := 1; i <= 100; i++ {
+		crd := fakeCRD("anvil", fmt.Sprintf("acme-%d.com", i))
+		if err := nt.KubeClient.Create(crd); err != nil {
+			nt.T.Fatal(err)
+		}
+	}
+
+	nt.T.Log("restarting the reconciler Pod to discovery the new 100 CRDs")
+	nomostest.DeletePodByLabel(nt, "app", reconcilermanager.Reconciler, true)
+	nt.T.Logf("Wait for the sync to complete")
+	if err := nt.WatchForAllSyncs(); err != nil {
```

Comment:
I'm not sure this will actually do anything. It's probably already synced. Replacing the reconciler may not cause it to become Synced=False. It will cause it to become Reconciling=True, but I don't know if WatchForAllSyncs checks for that. It might be good to make a commit afterwards to ensure it syncs something.

Reply:
Yes, you're right, `WatchForAllSyncs` doesn't check the reconciling condition.

Added a dummy commit to trigger the sync status change.